internal/daemon/plugin/vendorplugin.go

```diff
@@ -68,6 +138,18 @@ func NewGrpcPlugin(dpuMode bool, opts ...func(*GrpcPlugin)) *GrpcPlugin {
 		opt(gp)
 	}
 
+	if gp.vspImage != "" {
+		gp.log.Info("VSP Image", "vspImage", gp.vspImage)
+		data := gp.createCommonData(gp.vspImage)
+
+		err := gp.applyAllFromBinData(gp.log, "vsp-ds", data, client)
+		if err != nil {
+			gp.log.Error(err, "Failed to start vendor plugin container", "vspImage", gp.vspImage)
+		}
```

Comment:
Move into `WithVspImage`. Now, the logic is spread.

Reply:
Yeah I was debating this...I wasn't sure whether it was cleaner to have withVsp only concerned with the single task of setting the vsp, rather than having it also handling deploying a new container....

Originally I was going to add everything there...but open to doing it either way if you think it would be better to have it all logically contained in WithVspImage